Use Promise.all() instead of cockpit.all()

martinpitt opened the pull request:
This was fixed properly in Cockpit 186. The earliest version that we
still support anywhere (RHEL 7.7) is 195, so it is safe to use
Promise.all() now. To ensure this, bump the manifest's cockpit
dependencies accordingly.

This still leaves some cases where `cockpit.all()` is being used to
deliver promises with a progress() method. These need to wait until the
pages get ported to React.

Gundersanne commented:
The firewall failures (and maybe some of the others) on these are related to the .then() after a lot of the cockpit.all promises. cockpit.all returns each resolved promise as an argument to resolve() rather than in an array like Promise.all. So the bit of code that converts the arguments into a list can be dropped now I think.

martinpitt commented:
Thanks @Gundersanne! I took out the `results = Array.prototype.slice.call(arguments)` hacks, which fixed all firewall and (hopefully) most storage failures. There's still some left, but I don't currently spend much time on this.

martinpitt commented:
I fixed most of remaining test case failures. I reverted back to cockpit.all() in pkg/machines/libvirt-dbus.js `DELETE_VM`, as Promise.all() causes this [Error: Actions must be plain objects](https://209.132.184.41:8493/logs/pull-12795-20190918-111010-1e9eca60-cockpit-project-cockpit--fedora-30/log.html#69-2). @KKoukiou , maybe you have an idea why this passes the result of `Promise.all(storageVolPromises)`, as that code is supposed to return the chained `.then(() => call...` instead.

But this can be fixed in a follow-up. However, the test still fails for me locally, I'll look at that next week when I'm back.

martinpitt commented:
Trying to figure out if the [edge](https://209.132.184.41:8493/logs/pull-12795-20190919-063636-0d415e09-cockpit-project-cockpit--fedora-30-selenium-edge/log.html) test is some weird Edge specific regression, or just how broken it currently is. Retrying once.

martinpitt commented:
Adding needswork, it seems this oopses the Machines page with Edge. Other Edge tests look very sad as well, but they fail differently.

martinpitt commented:
Nevermind, not a regression from this PR. E. g. the edge test from [12798](https://209.132.184.41:8493/logs/pull-12798-20190920-074343-a8d82c89-cockpit-project-cockpit--fedora-30-selenium-edge/log.html#34) looks the same.
